tmt/utils.py

```diff
@@ -3600,3 +3670,136 @@ def __init__(
 
         kwargs.setdefault('logger', logger)
         super().__init__(node=node, **kwargs)
+
+
+FieldCLIOption = Union[str, Sequence[str]]
+
+
+@overload
+def field(
+        *,
+        default: bool,
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = True,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        # normalize: Optional[NormalizeCallback[T]] = None
+        ) -> bool:
+    pass
+
+
+@overload
+def field(
+        *,
+        default: T,
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = False,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        normalize: Optional[NormalizeCallback[T]] = None
+        ) -> T:
+    pass
+
+
+@overload
+def field(
+        *,
+        default_factory: Callable[[], T],
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = False,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        normalize: Optional[NormalizeCallback[T]] = None
+        ) -> T:
+    pass
+
+
+# TODO: ignore[misc]: mypy reports implementation cannot handle all possible arguments of
+# signatures 2 and 3, but I can't find the issue :/
+def field(  # type: ignore[misc]
+        *,
+        default: Any = dataclasses.MISSING,
+        default_factory: Any = dataclasses.MISSING,
+        # Options
+        option: Optional[FieldCLIOption] = None,
+        is_flag: bool = False,
+        choices: Union[None, Sequence[str], Callable[[], Sequence[str]]] = None,
+        multiple: bool = False,
+        metavar: Optional[str] = None,
+        help: Optional[str] = None,
+        # Input data normalization
+        normalize: Optional[NormalizeCallback[T]] = None
+        ) -> Any:
+    """
+    Define a :py:class:`DataContainer` field.
+
+    Effectively a fancy wrapper over :py:func:`dataclasses.field`, tailored for
+    tmt code needs and simplification of various common tasks.
+
+    :param defult: default value of the field.
```

Comment:
`defult`

Reply:
maybe whole param is redundant

Reply:
Correct, it was some outdated comment. Removed.